x/sync/client.go

```diff
@@ -205,20 +216,31 @@ func getAndParse[T any](ctx context.Context, client *client, request []byte, par
 		client.log.Debug("request failed, retrying",
 			zap.Stringer("nodeID", nodeID),
 			zap.Int("attempt", attempt),
-			zap.Error(err))
+			zap.Error(err),
+		)
 
 		if err != ctx.Err() {
-			// if [err] is being propagated from [ctx], avoid overwriting [lastErr].
+			// Don't overwrite [lastErr] if it's context cancelation.
 			lastErr = err
-			time.Sleep(failedRequestSleepInterval)
```

Comment:
Removed to avoid sleeping. It's not the end of the world to wait 10 ms to wake up in the event of context cancellation, but still seems like good practice to not sleep.